sticker/src/tensorflow/lr.rs

```diff
@@ -66,23 +81,34 @@ impl ExponentialDecay {
         );
 
         ExponentialDecay {
+            lr: initial_lr,
             initial_lr,
             decay_rate,
-            decay_steps,
+            decay_epochs: decay_steps,
             staircase,
+            warmup_steps,
         }
     }
 }
 
 impl LearningRateSchedule for ExponentialDecay {
-    fn learning_rate(&mut self, epoch: usize, _last_score: f32) -> f32 {
+    fn compute_step_learning_rate(&mut self, global_step: usize) -> f32 {
+        if global_step < self.warmup_steps {
+            (self.initial_lr / (self.warmup_steps as f32)) * global_step as f32
+        } else {
+            self.lr
+        }
+    }
+
+    fn compute_epoch_learning_rate(&mut self, epoch: usize, _last_score: f32) -> f32 {
         let exponent = if self.staircase {
-            (epoch / self.decay_steps) as f32
+            (epoch / self.decay_epochs) as f32
```

Comment:
For after this PR: once decay is also in batches/steps, we should probably start computing the exponent as `(steps - warmup_steps) / decay_steps`, so that decay only starts after the warmup.